sdk/cosmos/azure_data_cosmos/src/routing/collection_cache.rs

```diff
@@ -0,0 +1,271 @@
+                response.into_model()
+            })
+            .await
+    }
+
+    /// Removes container metadata from the cache, forcing refresh on next access.
+    ///
+    /// # Summary
+    /// Invalidates the cached container properties for the specified container ID.
+    /// The next call to `resolve_by_id` for this container will fetch fresh metadata
+    /// from the service. Useful when container configuration changes (e.g., partition
+    /// key updates, indexing policy modifications) and stale cache must be cleared.
+    ///
+    /// # Arguments
+    /// * `container_id` - Unique identifier of the container to remove from cache
+    pub async fn remove_by_id(&self, container_id: &str) {
+        self.container_properties_cache
+            .remove(&container_id.to_string())
+            .await;
+    }
+
+    /// Fetches container properties directly from the Cosmos DB service.
+    ///
+    /// # Summary
+    /// Executes an HTTP GET request to retrieve container metadata from the service.
+    /// Resolves the appropriate regional endpoint using the global endpoint manager,
+    /// constructs the request with proper routing context, and delegates to the retry
+    /// handler for resilient execution with automatic failover on errors.
+    ///
+    /// # Arguments
+    /// * `container_link` - Resource link identifying the target container
+    /// * `options` - Optional request options including consistency level and context
+    ///
+    /// # Returns
+    /// `Ok(Response<ContainerProperties>)` on success, or `Err` if request fails
+    async fn read_container_properties_by_id(
+        &self,
+        container_link: ResourceLink,
+        options: Option<ReadContainerOptions<'_>>,
+    ) -> azure_core::Result<Response<ContainerProperties>> {
+        let options = options.unwrap_or_default();
+        let mut cosmos_request =
+            CosmosRequest::builder(OperationType::Read, container_link.clone()).build()?;
+
+        let location_endpoint = Some(
+            self.global_endpoint_manager
+                .resolve_service_endpoint(&cosmos_request),
+        );
+
+        if let Some(ref endpoint) = location_endpoint {
+            cosmos_request
+                .request_context
+                .route_to_location_endpoint(cosmos_request.resource_link.url(endpoint));
+        }
+
+        let ctx_owned = options
+            .method_options
+            .context
+            .with_value(container_link)
+            .into_owned();
+
+        // Prepare a callback delegate to invoke the http request.
+        let sender = move |req: &mut CosmosRequest| {
+            let mut raw_req = req.clone().into_raw_request();
+            let ctx = ctx_owned.clone();
+            async move { self.pipeline.send(&ctx, &mut raw_req, None).await }
+        };
+
+        // Delegate to the retry handler, providing the sender callback
+        let res = self.retry_handler.send(&mut cosmos_request, sender).await;
```

Comment:
Layering and naming of the retry_handler is getting a little messy, but I'll hold on that for now. It's not really just about retry, it's our transport pipeline.